yeat/cli/auto_pop.py

```diff
@@ -0,0 +1,112 @@
+
+    def get_samples(self, samples):
+        if len(samples) == 1 and Path(samples[0]).is_file():
+            with open(samples[0], "r") as fh:
+                samples = fh.read().strip().split("\n")
+        return list(sorted(samples))
+
+    def check_samples(self):
+        for s1_index, s1 in enumerate(self.samples):
+            for s2_index, s2 in enumerate(self.samples):
+                if s1 == s2 and s1_index == s2_index:
+                    continue
+                if s1 in s2:
+                    message = f"cannot correctly process a sample name that is a substring of another sample name: {s1} vs. {s2}"
+                    raise AutoPopError(message)
+
+    def check_seq_path(self, seq_path):
+        if not seq_path:
+            return
+        if not Path(seq_path).exists():
+            message = f"path does not exist: '{seq_path}'"
+            raise AutoPopError(message)
+
+    def get_files(self, files, seq_path):
+        if files:
+            return [Path(f) for f in files]
+        files = []
+        for file in self.traverse(seq_path):
+            if not file.name.endswith(EXTENSIONS):
+                continue
+            files.append(file)
+        return files
+
+    def check_files(self):
+        for file in self.files:
+            if not file.exists():
+                message = f"file does not exist: {file}"
+                raise AutoPopError(message)
+
+    def traverse(self, dirpath):
+        dirpath = Path(dirpath)
+        if not dirpath.is_dir():
+            return  # pragma: no cover
+        for subpath in dirpath.iterdir():
+            if subpath.is_dir():
+                yield from self.traverse(subpath)
+            else:
+                yield subpath
+
+    def organize_files_to_samples(self):
+        files_to_samples = defaultdict(list)
+        for sample in self.samples:
+            files_to_samples[sample] = [file for file in self.files if sample in str(file)]
+        return files_to_samples
+
+    def check_files_to_samples(self):
+        for sample, files in self.files_to_samples.items():
+            if len(files) != 2:
+                message = f"sample {sample}: expected 2 FASTQ files for paired-end data, found {len(files)}"
+                raise AutoPopError(message)
+
+    def write_config_file(self, outfile):
+        data = self.get_config_data()
+        outdir = Path(outfile).parent
+        outdir.mkdir(parents=True, exist_ok=True)
+        outfile = open(outfile, "w")
+        json.dump(data, outfile, indent=4)
+
+    def get_config_data(self):
+        samples = {}
+        for label, reads in self.files_to_samples.items():
+            samples[label] = {"paired": [[str(read.absolute()) for read in reads]]}
+        assemblies = {
+            "spades-default": {
+                "algorithm": "spades",
+                "extra_args": "",
+                "samples": self.samples,
+                "mode": "paired",
+            }
+        }
```

Comment:
All samples will be dumped into the `spades-default` assembly since the auto-populate feature is only for paired-end reads.